script/testing/oltpbench/README.md

```diff
@@ -1,18 +1,94 @@
 # OLTP Benchmark Testing
 This folder include the necessary files for running oltpbench testings.
 
-# OLTP benchmark types
+## OLTP benchmark types
 Currently we run the following OLTP benchmarks
 - TATP
 - TPCC
 - NOOP
 - YCSB
 - Smallbank
 
-## External dependencies
+## How to run it?
+
+### TL;DR
+```bash
+# This command is sufficient for you to run the OLTP benchmark testing based on your configuration
+python3 run_oltpbench.py --config-file=<your-config-file-path>.json
+```
+
+### Examples
+Currently there are 3 types of the OLTP benchmark testing
+- End-to-end debug
+  ```bash
+  cd noisepage/script/testing/oltpbench
+  # To run the TPCC oltpbench test for end-to-end debug on debug version
+  python3 run_oltpbench.py \
+    --config-file=configs/end_to_end_debug/tpcc.json \
+    --build-type=debug
+  ```
+- End-to-end performance
+  ```bash
+  cd noisepage/script/testing/oltpbench
+  # To run the TPCC oltpbench test for end-to-end performance with wal to be stored in ramdisk on release version
+  python3 run_oltpbench.py \
+    --config-file=configs/end_to_end_performance/tpcc_wal_ramdisk.json \
+    --build-type=release
+  ```
+- Nightly performance
+  ```bash
+  cd noisepage/script/testing/oltpbench
+  # To run all the oltpbench test case for nightly performance with wal disabled on release version
+  python3 run_oltpbench.py \
+    --config-file=configs/nightly/nightly_wal_disabled.json \
+    --build-type=release
+  ```
+
+### Advanced options
+For more configurations, there are 2 ways
+- Via config files
+  - You can refer to the [config files](https://github.com/cmu-db/noisepage/blob/master/script/testing/oltpbench/README.md#config-files) section below for more details
+- Via command line options
+  - You can use `-h` or `--help` command line option for more details
+  - *In theory, if the same config are conflicting between config files and command line options, the config from __command line option should prevail__*
+
+## Test workflow
+- Start by invoking `run_oltpbench.py`
+  - Load the config file passed in and start an instance of the database with any server_args specified in the config
+  - Create a test suite for the whole config file and runs the test suite
+  - Create a `TestOLTPBench` object for the test execution
+    - Pass command line options and configs in to the constructor
+    - Try to locate the NoisePage binary
```

Comment:
I think this section could be a little confusing. I understand it pretty easily, but I think its because I am familiar with the code base. Explaining what a `TestOLTPBench` object is could help. When I read the sub-bullets I am not sure why this object does those things. We can either leave out the sub-bullets or explain why those things are done.

Reply:
explanation added, hopefully, it can help for the clarification